CDCgov/prime-reportstream
# Table Refactoring: Removes old pagination and splits out subcomponents

This PR is untangling the subcomponents of the RS Table component, and removes old pagination since it's no longer used anywhere. Test coverage is still good, **BUT** tests will be updated in a separate PR; they will test each component individually rather than all as a single Table test file.

There are three main places we use this table component:
- Daily Data & Details (FacilitiesTable is on detail page)
- Submission (not on Submission Details)
- ValueSets

Test Steps:
1. Ensure tables on the aforementioned pages are continuing to work as intended.
2. Run the test suite locally and ensure the table tests all pass.

## Changes
- Removes old pagination code from the Table component (subsequent PR will remove the files)
- Splits TableInfo, TableHeaders, and TableActionButton out from the main Table component code

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
